src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -106,7 +122,12 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return _method.ToUpperInvariant().GetHashCode();
+            if (_hashcode == 0)
+            {
+                _hashcode = _method.ToUpperInvariant().GetHashCode();
```

Comment:
I think that is overly complicated in trying to sub-optimize something that it not really that critical in the overall cost of devs using these APIs.  It is very rare to even see custom HttpMethods at all.  Most just use `HttpMethod.Get`, `HttpMethod.Post`, 'HttpMethod.Head`, etc.